src/read_params.jl

```diff
@@ -0,0 +1,46 @@
+
+# ------------------------------------------
+#   Read ACE fit parameters from file 
+
+using ACE1pack
+
+export  fill_defaults!
+
+""" Recursively updates nested dictionaries with default parameters"""
+function fill_defaults!(params::Dict; param_key = "fit_params")
+    # Go through the nested dictionaries filling in the default values
+    params = _fill_default(params, param_key)
+    for (key, val) in params
+        if val isa Dict &&  ~(key in ["weights", "e0"])
+            params[key] = fill_defaults!(val; param_key = key)
+        elseif  key == "basis"
```

Comment:
Maybe it would be better to organise this as 

```
fit_params["basis"] = Dict(
    "pair_basis" => pair_basis_param_dict,
    "rpi_bsasis" => rpi_basis_param_dict) 
```
instead of the following?
```
fit_params["basis"] = [
    "pair_basis" => pair_basis_param_dict,
    "rpi_bsasis" => rpi_basis_param_dict]
```

The former would make it easier to fill in, the latter is a bit more general.

Reply:
Actually it is not really more general, we can always loop 
```
for (basisname, basisdict) in fit_params["basis"]
    ... 
end
```

Reply:
The problem with the second approach is that it is not "raw" JSON.

Reply:
Of course, it's also trivial to iterate over a dictionary.. Updated.